metadata/src/test/java/org/apache/kafka/controller/ReplicationControlManagerTest.java

```diff
@@ -605,6 +617,16 @@ public void testCreateTopicsWithConfigs() throws Exception {
             "Null value not supported for topic configs: foo",
             result2.response().topics().find("bar").errorMessage()
         );
+
+        CreateTopicsRequestData request3 = new CreateTopicsRequestData();
+        request3.topics().add(new CreatableTopic().setName("baz")
+            .setNumPartitions(-1).setReplicationFactor((short) -2)
+            .setConfigs(validConfigs));
+
+        ControllerResult<CreateTopicsResponseData> result3 =
+            replicationControl.createTopics(request3, Collections.singleton("baz"));
+        assertEquals(INVALID_REPLICATION_FACTOR.code(), result3.response().topics().find("baz").errorCode());
+        assertEquals(0, result3.records().size());
```

Comment:
nit: a better assertion is this:
```java
assertEquals(Collections.emptyList(), result3.records());
```
It gives more information in case of a failure.